src/rai/rai/utils/model_initialization.py

```diff
@@ -0,0 +1,73 @@
+# Copyright (C) 2024 Robotec.AI
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Literal, TypedDict
+
+import tomli
+
+
+class LLMConfig(TypedDict):
+    vendor: str
+    simple_model: str
+    complex_model: str
+    region_name: str
+    base_url: str
+
+
+class EmbeddingsConfig(TypedDict):
+    vendor: str
+    model: str
+    region_name: str
+    base_url: str
```

Comment:
Isn't dataclass better for this usecase? Besides accessing the dict/class keys (which comes to personal preference - my is to access `config.vendor` rather than `config["vendor"]`), using dicts might be error prone. Please consider not-full config case:
```
config = EmbeddingsConfig(dict(vendor='abc'))
```
then `model` is missing and will raise KeyError when we try to initialize embeddings.
With dataclass, we ensure that all attributes are set once config is created - so probably at the very beginning of the program

Reply:
Makes sense. I've updated the code.